server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/CassandraBlobsDAO.java

```diff
@@ -47,24 +44,23 @@
 import org.apache.james.blob.api.ObjectStoreException;
 import org.apache.james.blob.cassandra.BlobTable.BlobParts;
 import org.apache.james.blob.cassandra.utils.DataChunker;
-import org.apache.james.util.FluentFutureStream;
-import org.apache.james.util.OptionalUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.apache.james.blob.cassandra.utils.PipedStreamSubscriber;
 
 import com.datastax.driver.core.PreparedStatement;
 import com.datastax.driver.core.Row;
 import com.datastax.driver.core.Session;
 import com.github.fge.lambdas.Throwing;
-import com.github.fge.lambdas.consumers.ConsumerChainer;
-import com.github.steveash.guavate.Guavate;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableList;
 import com.google.common.primitives.Bytes;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
 
 public class CassandraBlobsDAO implements BlobStore {
-    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraBlobsDAO.class);
+
+    private static final int PREFETCH = 16;
+    private static final int MAX_CONCURRENCY = 2;
```

Comment:
how does it behave on a real loaded server? Is this per session/connection or for the whole server?

Reply:
Gatling results are quite good.
The mean of this is: we let reactor use as much as 2 concurrent processes to interact with Cassandra.
As it uses a system-wide Scheduler (elastic), if it can get enough threads, it won't be concurrent.
BTW, reactor is very frugal in term of threads so most of the time, it will probably be able to honor concurrency request